src/social-network/ui.ts

```diff
@@ -87,22 +88,27 @@ export interface SocialNetworkUIInjections {
     injectPostBox(): void
     /**
      * This function should inject the Welcome Banner
+     * @desc why both undefined and null
+     * undefined means undefined (at definition time),
+     * everything must be defined at runtime
+     * (for the convenient of managing config items),
+     * so the null explicit defines "not available".
```

Comment:
is ```nop``` better?

Reply:
yes, choose meaningful words for each place.

Reply:
![image](https://user-images.githubusercontent.com/15797507/65833687-6a57ee80-e305-11e9-8319-6fd1b53c71ad.png)
Using literal string instead of null will result use of any and I had working for over one hour on this to try to avoid any. Is it okay?

Reply:
The thing is, ```null``` is ```null``` but ```'disabled'``` is ```string```.

Reply:
the type can be set to `whateverFunctionType | 'disabled'` to accept a single exact string.

Reply:
If that we had to duplicate keys we want.

Reply:
In value level, use `'disabled' as const`, in type level, use `type Disabled = 'disabled'`

Reply:
@Misaka-0x447f Does this get resolved?